responder/http_responder_test.go

```diff
@@ -0,0 +1,29 @@
+package responder
+
+import (
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/nicksdlc/macaw/config"
+	"github.com/nicksdlc/macaw/connectors"
+)
+
+func TestShouldRespondWithPreSetResponseToMessage(t *testing.T) {
+	expected := "\"name\": 10"
+
+	sut := NewMessageResponder(connectors.NewHTTPConnector("/test", "GET", 12345, nil), config.Response{Amount: 1})
```

Comment:
Using hardcoded port in tests is dangerous. On some build machine or even your machine test will fail if the port is already in use

Reply:
You can add a new constructor like "WithHTTPListener" and pass the test server (if they have same type, not sure).
Or create something like NewTestConnector which uses a test HTTP server under the hood
Or create a function that checks if a port in use before creating a connector